Admin/Configuration/AdminConfiguration.php

```diff
@@ -2,190 +2,68 @@
-        $this->dataProvider = $configuration['data_provider'];
-        $this->metadata = $configuration['metadata'];
-    }
-
-    /**
-     * @return string
+     * @param ApplicationConfiguration $applicationConfiguration
      */
-    public function getControllerName()
+    public function __construct(ApplicationConfiguration $applicationConfiguration)
     {
-        return $this->controllerName;
-    }
+        parent::__construct();
 
-    /**
-     * @return string
-     */
-    public function getEntityName()
-    {
-        return $this->entityName;
+        $this->applicationConfiguration = $applicationConfiguration;
     }
 
     /**
-     * @return string
-     */
-    public function getFormType()
-    {
-        return $this->formType;
-    }
-
-    /**
-     * @return array
-     */
-    public function getActions()
-    {
-        return $this->actions;
-    }
-
-    /**
-     * @return int
-     */
-    public function getMaxPerPage()
-    {
-        return $this->maxPerPage;
-    }
-
-    /**
-     * @return string
-     */
-    public function getRoutingNamePattern()
-    {
-        return $this->routingNamePattern;
-    }
-
-    /**
-     * @param string $routingNamePattern
-     */
-    public function setRoutingNamePattern($routingNamePattern)
-    {
-        $this->routingNamePattern = $routingNamePattern;
-    }
-
-    /**
-     * @return string
-     */
-    public function getRoutingUrlPattern()
-    {
-        return $this->routingUrlPattern;
-    }
-
-    /**
-     * @param string $routingUrlPattern
-     */
-    public function setRoutingUrlPattern($routingUrlPattern)
-    {
-        $this->routingUrlPattern = $routingUrlPattern;
-    }
-
-    /**
-     * @return ClassMetadata
-     */
-    public function getMetadata()
-    {
-        return $this->metadata;
-    }
-
-    /**
-     * Return custom data provider name.
-     *
-     * @return string
-     */
-    public function getDataProvider()
-    {
-        return $this->dataProvider;
-    }
-
-    /**
-     * Return original admin configuration.
-     *
-     * @return array
+     * @param OptionsResolver $resolver
      */
-    public function getAdminConfiguration()
+    public function configureOptions(OptionsResolver $resolver)
     {
-        return $this->adminConfiguration;
+        // inherited routing configuration from global application configuration
+        $routing = $this
+            ->applicationConfiguration
+            ->getParameter('routing');
+
+        // inherited max per page configuration
+        $maxPerPage = $this
+            ->applicationConfiguration
+            ->getParameter('max_per_page');
+
+        // optional options
+        $resolver->setDefaults([
+            'actions' => [
+                'list' => [],
+                'create' => [],
+                'edit' => [],
+                'delete' => [],
+            ],
+            'batch' => true,
+            'manager' => 'LAG\AdminBundle\Manager\GenericManager',
```

Comment:
depending on the minimum version of PHP, you might consider using the ::class notation